R/importNet.Multinet.R

```diff
@@ -0,0 +1,145 @@
+#' @param verbose Controls whether to print summary information after import. 
+#'   Defaults to TRUE.
+#' @param silent Controls whether output to console should be suppressed. FALSE
+#'   by default.
+#' @param writeLog Controls whether writing the result to the global logfile is
+#'   enabled. TRUE by default.
+#' @return a weighted, directed, simple igraph graph which is a rete gG object.
+#'
+#' @family ImportNet.STRING, importNet.MultiNet, importNet.MITAB
+#'
+#'   ## @seealso \code{\link{fastMap}} fastMap() is used internally to map ENSP
+#'   IDs to gene symbols.
+#'
+#'   ## @examples ## \dontrun{ ## importNet.MULTINET(IN, OUT) ## }
+#' @export
+importNet.MultiNet <- function(fName,
+                             net = "PPI",
+                             verbose = TRUE,
+                             silent = FALSE,
+                             writeLog = TRUE) {
+
+    # ==== VALIDATIONS =========================================================
+
+    # General parameter checks
+    cR <- character()
+    cR <- c(cR, .checkArgs(fName,        like = "FILE_E",   checkSize = TRUE))
+    cR <- c(cR, .checkArgs(net,          like = "a",        checkSize = TRUE))
+    cR <- c(cR, .checkArgs(verbose,      like = logical(1), checkSize = TRUE))
+  
+    if(length(cR) > 0) {
+        stop(cR)
+    }
+    
+    # Read header and one contents line
+    tmp <- readLines(fName, n = 2)
+    
+    data   <- unlist(strsplit(tmp[2], " "))
+    
+    # Parse for requested column. STRING data is " "-delimited.
+    header <- unlist(strsplit(tmp[1], " "))
+    iCol <- which(net == header)  # column that contains the requested net
+    
+    #  Validate that requested network exists in data
+    if (length(iCol) != 1) {
+        stop(paste0("Request error:", "\n",
+                    "Requested network type is ", sprintf("\"%s\"", net), "\n",
+                    "Found network type(s): <",
+                    paste(header[3:length(header)], collapse = ", "),
+                    ">", "\n"))
+    }
+
+    # ==== READ DATA ===========================================================
+
+    # Create column mask for readr::read_delim()
+    readMask <- paste("cc",
+                      paste(rep("_", length(header) - 2), collapse = ""),
+                      sep = "")
+    substr(readMask, iCol, iCol) <- "i"   # set target column to integer
+    
+    # start reading
+    if (!silent) { cat("Reading", net, "from", fName, "...") }
+    
+    # Read file into data frame, use only selected column
+    netDF <- readr::read_delim(file = fName,
+                               delim = " ",
+                               col_types = readMask,
+                               n_max = Inf)
+    
+    colnames(netDF) <- c("a", "b", "weight")
+    
+    nIn <- nrow(netDF)
+    if (!silent) { cat("done. (", nIn, " edges)", "\n") }
+    
+    
+    # ==== CREATE GRAPH OBJECT =================================================
+    
+    gG <- .df2gG(netDF)
```

Comment:
gG is a directed graph. MultiNet has undirected edges. You need to do something about that. What?